src/epcpm/sunspectointerface.py

```diff
@@ -0,0 +1,2227 @@
+                )
+                rows_out.append(
+                    f"}} Sunspec{self.sunspec_id.value}Model{self.model_id}_{parameter.abbreviation};\n\n"
+                )
+                return rows_out, default_enumerator
+
+            return rows_out, default_enumerator
+
+        enumeration = self.parameter_uuid_finder(enumeration_uuid)
+
+        enumerators_by_bit = {
+            enumerator.value: enumerator for enumerator in enumeration.children
+        }
+
+        return _output_enum_or_bitfield(
+            self.wrapped.size,
+            self.sunspec_id,
+            self.model_id,
+            sunspec_type,
+            enumerators_by_bit,
+            parameter,
+        )
+
+
+@enumeration_builders(epcpm.sunspecmodel.DataPointBitfield)
+@attr.s
+class EnumerationDataPointBitfield:
+    """Enumeration interface generator for the SunSpec DataPointBitfield class."""
+
+    wrapped = attr.ib()
+    model_id = attr.ib()
+    parameter_uuid_finder = attr.ib(default=None)
+    sunspec_id = attr.ib(default=None)
+
+    def gen(
+        self,
+    ) -> typing.Tuple[
+        typing.List[str], typing.Dict[str, epyqlib.pm.parametermodel.SunSpecEnumerator]
+    ]:
+        """
+        Enumeration interface generator for the SunSpec DataPointBitfield class.
+
+        Returns:
+            [list[str], dict[str, SunSpecEnumerator]: list of output, dict of SunSpecEnumerator's
+        """
+        parameter = self.parameter_uuid_finder(self.wrapped.parameter_uuid)
+
+        sunspec_type = self.parameter_uuid_finder(self.wrapped.type_uuid).name
+
+        # 16 bits per register
+        total_bit_count = self.wrapped.size * 16
+        decimal_digits = len(str(total_bit_count - 1))
+
+        enumerators_by_bit = {}
+        bit = 0
+        for child in self.wrapped.children:
+            child_parameter = self.parameter_uuid_finder(child.parameter_uuid)
+
+            if child.bit_length > 1:
+                for child_bit in range(child.bit_length):
+                    padded_bit_string = f"{child_bit:0{decimal_digits}}"
+                    enumerator = epyqlib.pm.parametermodel.SunSpecEnumerator(
+                        abbreviation=f"{child_parameter.abbreviation}{padded_bit_string}",
+                        value=bit,
+                    )
+                    enumerators_by_bit[bit] = enumerator
+                    bit += 1
+            else:
+                enumerator = epyqlib.pm.parametermodel.SunSpecEnumerator(
+                    abbreviation=f"{child_parameter.abbreviation}",
+                    value=bit,
+                )
+                enumerators_by_bit[bit] = enumerator
+                bit += 1
+
+        return _output_enum_or_bitfield(
+            self.wrapped.size,
+            self.sunspec_id,
+            self.model_id,
+            sunspec_type,
+            enumerators_by_bit,
+            parameter,
+        )
+
+
+def _output_enum_or_bitfield(
```

Comment:
Here is another function that seems hard to maintain and should be broken out a bit.

Reply:
There was some effort to break this out. Eventually it just became a docstring mess. I'll revisit another day.